typescript/agentkit/src/action-providers/yelay/yelayActionProvider.ts

```diff
@@ -0,0 +1,356 @@
+import { ActionProvider } from "../actionProvider";
+import { Network } from "../../network";
+import { CreateAction } from "../actionDecorator";
+import { EvmWalletProvider } from "../../wallet-providers";
+import {
+  YelayClaimSchema,
+  YelayDepositSchema,
+  YelayRedeemSchema,
+  YelayBalanceSchema,
+} from "./schemas";
+import type { APYResponse, VaultsDetailsResponse, ClaimRequest, ChainId } from "./types";
+import {
+  CONTRACTS_BY_CHAIN,
+  RETAIL_POOL_ID,
+  YELAY_BACKEND_URL,
+  YELAY_VAULT_ABI,
+  YIELD_EXTRACTOR_ABI,
+} from "./constants";
+import { parseUnits, encodeFunctionData, formatUnits } from "viem";
+
+import { approve } from "../../utils";
+
+// Mainnet, Sonic, Base, Arbitrum, Avalanche
+const SUPPORTED_CHAIN_IDS = ["1", "146", "8453", "42161", "43114"];
+/**
+ * YelayActionProvider provides actions for yelay operations.
+ *
+ * @description
+ * This provider is designed to work with EvmWalletProvider for blockchain interactions.
+ * It supports all evm networks.
+ */
+export class YelayActionProvider extends ActionProvider<EvmWalletProvider> {
+  /**
+   * Constructor for the YelayActionProvider.
+   */
+  constructor() {
+    super("yelay", []);
+  }
+
+  /**
+   * Gets the details of the Yelay vaults with their last week APY.
+   *
+   * @param wallet - The wallet instance to get chainId
+   * @returns A formatted string containing the list of vaults with their APY.
+   */
+  @CreateAction({
+    name: "get_vaults",
+    description: `Getting Yelay vaults for the base network`,
+    schema: z.object({}),
+  })
+  async getVaults(wallet: EvmWalletProvider): Promise<string> {
+    let vaultsResponse: Response;
+    let vaultAPYsResponse: Response;
+
+    try {
+      const chainId = wallet.getNetwork().chainId!;
+      [vaultsResponse, vaultAPYsResponse] = await Promise.all([
+        fetch(`${YELAY_BACKEND_URL}/vaults?chainId=${chainId}`),
+        fetch(`${YELAY_BACKEND_URL}/interest/vaults?chainId=${chainId}`),
+      ]);
+
+      if (!vaultsResponse.ok || !vaultAPYsResponse.ok) {
+        const errorMessage = !vaultsResponse.ok
+          ? `Failed to fetch vaults: ${vaultsResponse.status} ${vaultsResponse.statusText}`
+          : `Failed to fetch APYs: ${vaultAPYsResponse.status} ${vaultAPYsResponse.statusText}`;
+        throw new Error(errorMessage);
+      }
+
+      const [vaults, vaultAPYs] = await Promise.all([
+        vaultsResponse.json() as Promise<VaultsDetailsResponse[]>,
+        vaultAPYsResponse.json() as Promise<APYResponse[]>,
+      ]);
+
+      const vaultsDetails = vaults.map(vault => ({
+        ...vault,
+        apy: vaultAPYs.find(apy => apy.vault === vault.address)?.apy,
+      }));
+
+      return vaultsDetails
+        .map(
+          vault => `
+${vault.name}:
+Address: ${vault.address}
+APY: ${vault.apy}%
+`,
+        )
+        .join("----------------");
+    } catch (error) {
+      if (error instanceof Error) {
+        console.error("Error fetching vault data:", error.message);
```

Comment:
nit: we usually don't have console logs in actions. Please change the tests accordingly and make sure they pass. 
```suggestion
return "Error fetching vault data: " + error.message;
```